File: security/src/main/java/top/bettercode/summer/security/IResourceService.java
package top.bettercode.summer.security;

import java.util.Collections;
import java.util.List;

/**
 * @author Peter Wu
 */
public interface IResourceService {

  /**
   * @return 所有资源
   */
  default List<? extends IResource> findAllResources() {
    return Collections.emptyList();
  }

  default boolean supportsAnonymous() {
    return true;
  }
}

File: security/src/main/kotlin/top/bettercode/summer/security/IResourceService.kt
package top.bettercode.summer.security

/**
 * @author Peter Wu
 */
interface IResourceService {
    /**
     * @return 所有资源
     */
    @JvmDefault
    fun findAllResources(): List<IResource> {
        return emptyList()
    }

    @JvmDefault
    fun supportsAnonymous(): Boolean {
        return true
    }
}
